lib/diagnostic.js

```diff
@@ -8,21 +8,38 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+  let result = arguments[0];
+  for (let i = 0; i < arguments.length; i++) {
+   if(result > arguments[i]){
+     result = arguments[i];
+   }
+  }
+return result;
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
-const sum = function sum() {
-
+const sum = function sum(array) {
+  let result = 0;
+  for (let i = 0; i < arguments.length; i++) {
+    result = arguments[i] + result;
+  }
+return result;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
-const newArray = function newArray(/* length, defaultValue */) {
+const newArray = function newArray(arrayLength, defaultValue) {
+  let result = [];
+
+  for (let i = 0; i < arrayLength.length; i++) {
```

Comment:
You're a little confused here. What is `arrayLength` supposed to be? It's a length, meaning it is an integer. It is not an actual array itself.